python_modules/libraries/dagster-aws/dagster_aws/secretsmanager/resources.py

```diff
@@ -70,40 +108,127 @@ def example_job():
+
+    Note that your ops must also declare that they require this resource with or it will not be initialized
+    for the execution of their compute functions.
+    """
+
+    secrets: List[str] = Field(
+        default=[], description="An array of AWS Secrets Manager secrets arns to fetch."
     )
-)
+    secrets_tag: Optional[str] = Field(
+        default=None,
+        description="AWS Secrets Manager secrets with this tag will be fetched and made available.",
+    )
+
+    @contextmanager
+    def secrets_in_environment(
+        self,
+        secrets: Optional[List[str]] = None,
+        secrets_tag: Optional[str] = None,
+    ) -> Generator[Dict[str, str], None, None]:
+        """Yields a dict which maps selected SecretsManager secrets to their string values. Also
+        sets chosen secrets as environment variables.
+
+        Args:
+            secrets (Optional[List[str]]): An array of AWS Secrets Manager secrets arns to fetch.
+                Note that this will override the secrets specified in the resource config.
+            secrets_tag (Optional[str]): AWS Secrets Manager secrets with this tag will be fetched
+                and made available. Note that this will override the secrets_tag specified in the
+                resource config.
+        """
+        secrets_manager = construct_secretsmanager_client(
+            max_attempts=self.max_attempts,
+            region_name=self.region_name,
+            profile_name=self.profile_name,
+        )
+
+        secrets_tag_to_fetch = secrets_tag if secrets_tag is not None else self.secrets_tag
+        secrets_to_fetch = secrets if secrets is not None else self.secrets
+
+        secret_arns = merge_dicts(
+            (
+                get_tagged_secrets(secrets_manager, [secrets_tag_to_fetch])
+                if secrets_tag_to_fetch
+                else {}
+            ),
+            get_secrets_from_arns(secrets_manager, secrets_to_fetch),
+        )
+
+        secrets_map = {
+            name: secrets_manager.get_secret_value(SecretId=arn).get("SecretString")
+            for name, arn in secret_arns.items()
+        }
+        with environ(secrets_map):
+            yield secrets_map
+
+    def fetch_secrets(
+        self,
+        secrets: Optional[List[str]] = None,
+        secrets_tag: Optional[str] = None,
+    ) -> Dict[str, str]:
+        """Fetches secrets from AWS Secrets Manager and returns them as a dict.
+
+        Args:
+            secrets (Optional[List[str]]): An array of AWS Secrets Manager secrets arns to fetch.
+                Note that this will override the secrets specified in the resource config.
+            secrets_tag (Optional[str]): AWS Secrets Manager secrets with this tag will be fetched
+                and made available. Note that this will override the secrets_tag specified in the
+                resource config.
+        """
+        with self.secrets_in_environment(secrets=secrets, secrets_tag=secrets_tag) as secret_values:
```

Comment:
[1] 
maybe i'm missing something (and this isn't really the point of this pr to begin with) but wouldn't this method also end up setting all of the secrets as environment variables because it's calling `secrets_in_environment`?